test/HttpTest.cpp

```diff
@@ -71,15 +71,15 @@ TEST(HttpServer, HttpTest) {
         threads.emplace_back([&]() {
           for (size_t j = 0; j < 20; ++j) {
             {
-              HttpClient httpClient("localhost",
-                                    std::to_string(httpServer.getPort()));
-              ASSERT_EQ(toString(httpClient.sendRequest(verb::get, "localhost",
-                                                        "target1", handle)),
-                        "GET\ntarget1\n");
-              ASSERT_EQ(
-                  toString(httpClient.sendRequest(verb::post, "localhost",
-                                                  "target1", handle, "body1")),
-                  "POST\ntarget1\nbody1");
+              auto httpClient = std::make_unique<HttpClient>(
+                  "localhost", std::to_string(httpServer.getPort()));
+
+              auto response =
+                  HttpClient::sendRequest(std::move(httpClient), verb::get,
+                                          "localhost", "target1", handle);
+              ASSERT_EQ(response.status_, boost::beast::http::status::ok);
+              ASSERT_EQ(response.contentType_, "text/plain");
+              ASSERT_EQ(toString(std::move(response.body_)), "GET\ntarget1\n");
```

Comment:
If the HTTP server is some testServer that we have under our control,
Can you modify it such tat we can say something like
`httpServer.setStatusCodeForNextRequest()`
and the same for the `mediaType`
(then we can test that those can in fact change, which is always a good idea.